commec/config/io_parameters.py

```diff
@@ -0,0 +1,209 @@
+        self.fasta_aa_filepath = f"{output_prefix}.transeq.faa"
+        command = ["transeq", self.cleaned_fasta_filepath, self.fasta_aa_filepath, "-frame", "6", "-clean"]
+
+        # hard coded into subprocess.run for now, as have yet to decide where run_as_subprocess should be.
+        # TODO: Update this.
+        try:
+            result = subprocess.run(command, check=True)
+            if result.returncode != 0:
+                command_str = ' '.join(command)
+                raise RuntimeError(
+                    f"subprocess.run of command '{command_str}' encountered error."
+                )
+        except RuntimeError as error:
+            raise RuntimeError("Input FASTA {fasta_to_screen} could not be translated.") from error
+        
+    @staticmethod
+    def get_cleaned_fasta(input_file, out_prefix):
+        """
+        Return a FASTA where whitespace (including non-breaking spaces) and illegal characters are
+        replaced with underscores.
+        """
+        cleaned_file = f"{out_prefix}.cleaned.fasta"
+        with (
+            open(input_file, "r", encoding="utf-8") as fin,
+            open(cleaned_file, "w", encoding="utf-8") as fout,
+        ):
+            for line in fin:
+                line = line.strip()
+                modified_line = "".join(
+                    "_" if c.isspace() or c == "\xc2\xa0" or c == "#" else c for c in line
+                )
+                fout.write(f"{modified_line}{os.linesep}")
+        return cleaned_file
+
+class ScreenIOParameters():
+    """
+    Container for all input settings, and for all locations and directory structures of inputs / outputs.
+    Is constructed with the argsparser outputs, and validates all input settings, and creates secondary
+    input settings - such as cleaning fastas, and running transeq.
+    """
+
+    def __init__(self, args : argparse.ArgumentParser):
+        # Ensure required arguments are present
+        required_args = ['fasta_file', 'threads', 'protein_search_tool', 'fast_mode',
+                         'skip_nt_search', 'cleanup', 'output_prefix', 'database_dir', 'jobs']
+        for arg in required_args:
+            if not hasattr(args, arg):
+                raise ValueError(f"Missing required argument: {arg}")
+
+        # Inputs
+        self.inputs : ScreenInputParameters = ScreenInputParameters(
+            args.threads,
+            args.protein_search_tool,
+            args.fast_mode,
+            args.skip_nt_search,
+            args.cleanup,
+            args.jobs,
+        )
+
+        # TODO: Think about whether logs belong in here, or externally.
+
+        # Outputs
+        self.output_prefix = self.get_output_prefix(args.fasta_file, args.output_prefix)
+        self.output_screen_file = f"{self.output_prefix}.screen"
+        self.tmp_log = f"{self.output_prefix}.log.tmp"
+
+        #Query
+        self.query : Query = Query(args.fasta_file, f"{self.output_prefix}.transeq.faa")
```

Comment:
Can `Query` instead take an input and an output prefix in a constructor? I feel like only `Query` should need to know that we've chosen to call our translated file `f"{self.output_prefix}.transeq.faa")`

Reply:
Good catch, will do.